packages/gestalt-core/build.js

```diff
@@ -131,19 +132,35 @@ const plugins = (name) => [
   }),
   nodeResolve(),
   replace({
-    'process.env.NODE_ENV': JSON.stringify(process.env.NODE_ENV || 'development'),
+    preventAssignment: true,
+    values: {
+      'process.env.NODE_ENV': JSON.stringify(process.env.NODE_ENV || 'development'),
+    },
   }),
   svgPath(),
+
   json({
     preferConst: true,
   }),
   babel({
     babelrc: false,
     babelHelpers: 'bundled',
     presets: [['@babel/preset-env', { modules: false }], '@babel/react', '@babel/flow'],
-    plugins: ['@babel/proposal-class-properties'],
+    plugins: [
+      '@babel/proposal-class-properties',
+      [
+        process.env.NODE_ENV === 'development'
+          ? '@babel/plugin-transform-react-jsx-self'
+          : '@babel/plugin-transform-react-jsx',
+        {
+          runtime: 'automatic',
+          useBuiltIns: true,
+        },
+      ],
+    ],
     exclude: 'node_modules/**',
   }),
+  commonjs(),
```

Comment:
Needed for `react/jsx-runtime` which exports `jsx` on the module export.
Without it, the transformed import `import {jsx as _jsx} from 'react/jsx-runtime'` will fail.